### ISSUE TITLE
Intermediate fine-tuning (sequential transfer learning)?

### ISSUE THREAD

**Issue by higo522** [ultralytics/ultralytics#21750]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have a question about transfer learning for a domain that's very different from COCO. I want to build a model for detecting vehicles in top-down, thermal aerial imagery taken from an airplane. Since the features learned from the COCO dataset (natural, everyday images) is very different for my use case (thermal aerial urban imagery), my idea was to bridge this gap with a two-step fine-tuning approach: 

**1. Intermediate Pretraining**: first, fine-tune the standard COCO-pretrained model on a large, general-purpose aerial thermal dataset of any object (ex. humans, cars, animals, houses, taken from top-down view) to adapt the YOLO models from the COCO domain to the new thermal aerial domain. 
**2. Final Fine-tuning**: Then, use this resulting domain specific "thermal aerial foundation" model as the starting point for a second round of fine-tuning on my final, curated dataset with the target classes (thermal aerial imagery of urban neighborhoods) for my specific task (detecting vehicles from planes).

Is this "double pretraining" strategy a recommended, well-documented way to bridge a large domain gap (in literature)? If so, are there any best practices for the intermediate pretraining step, such as using a different learning rate or number of epochs compared to the final fine-tuning stage? Moroever, is there a recommended sample size or other considerations for the domain-specific general dataset used for intermediate pretrainin?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @higo522, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Great question—yes, your “intermediate pretraining → task fine-tuning” plan is a well-established way to bridge large domain gaps (often called domain‑adaptive pretraining/sequential transfer learning).

Practical guidance:
- Stage 1 (domain adaptation): use standard YOLO settings on a broad thermal/aerial corpus; prioritize diversity (altitude, sensors, seasons) over class balance; 5k–50k images is common, more is better.
- Stage 2 (task tuning): reduce lr 5–10× vs. Stage 1, optionally freeze early layers for stability on small curated sets (e.g., freeze=10 for initial epochs).
- Small objects: train at higher resolution (imgsz 1024–1280) and keep multiscale/augmentation conservative to preserve tiny targets.
- Thermal specifics: if images are single-channel, convert to 3-channel consistently before/within the loader to match model expectations.
- Use early stopping and LR scheduling; monitor target‑domain val during both stages.

Example CLI:
yolo detect train model=yolo11n.pt data=thermal_aerial_general.yaml imgsz=1024 epochs=100 batch=-1 lr0=0.01 lrf=0.01 patience=20
yolo detect train model=runs/detect/train/weights/best.pt data=vehicles_thermal_urban.yaml imgsz=1024 epochs=100 batch=-1 lr0=0.001 lrf=0.01 freeze=10 patience=30

For background and tuning tips, see the Transfer Learning overview, Fine‑tuning guide, and Model Training Tips, and for overhead imagery datasets, the xView dataset page:
- Transfer learning: https://www.ultralytics.com/glossary/transfer-learning
- Fine-tuning: https://www.ultralytics.com/glossary/fine-tuning
- Training tips (LR scheduling, patience, epochs): https://docs.ultralytics.com/guides/model-training-tips/
- xView dataset (overhead domain): https://docs.ultralytics.com/datasets/detect/xview/
